### PR TITLE
Fix mips.gnu did not judge Bit Endianness before instruction analysis ##anal 

### LINKED ISSUE #17413: anal_mips_gnu.c  Code analysis bug
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Ubuntu x86
| File format of the file you reverse (mandatory)      |  ELF.
| Architecture/bits of the file (mandatory)            | MIPS
| r2 -v full output, **not truncated** (mandatory)     | radare2 4.6.0-git 25030 @ linux-x86-64 git.4.4.0-504-g78ea6ec78
commit: 78ea6ec78c908ad71a57866664798225b0e8f62d build: 2020-08-07__16:11:30

### Expected behavior
```
: e asm.arch =mips.gnu
: V
: p
```
Disassembly panel 

You should display the analyzed annotation content as you would with capstone
```
            0x1200009cc      2dc88003       move t9, gp
            0x1200009d0      02001c3c       lui gp, 2
            0x1200009d4      44869c27       addiu gp, gp, -0x79bc
            0x1200009d8      2de09f03       daddu gp, gp, ra
            0x1200009dc      2df80000       move ra, zero
            0x1200009e0      208084df       ld a0, -main(gp)           ;# [0x120011030:8]=0x120000c88 sym.main
            0x1200009e4      0000a5df       ld a1, (sp)
            0x1200009e8      0800a667       daddiu a2, sp, 8
            0x1200009ec      f0ff0124       addiu at, zero, -0x10
            0x1200009f0      24e8a103       and sp, sp, at
            0x1200009f4      288087df       ld a3, -sym.__libc_csu_init(gp) ;# [0x120011038:8]=0x120000cf0 sym.__libc_csu_init
            0x1200009f8      308088df       ld t0, -sym.__libc_csu_fini(gp) ;# [0x120011040:8]=0x120000d88 sym.__libc_csu_fini
```

### Actual behavior
```
  ╎╎╎╎╎╎╎   0x1200009c0      2d00e003       move zero, ra              ; [10] -r-x section size 976 named .text
  ╎╎╎╎╎╎╎   0x1200009c4      01001104       bal 0x1200009cc
  ╎╎╎╎╎╎╎   0x1200009c8      00000000       nop
  ╎╎╎╎╎╎╎   0x1200009cc      2dc88003       move t9, gp
  ╎╎╎╎╎╎╎   0x1200009d0      02001c3c       lui gp, 0x2
  ╎╎╎╎╎╎╎   0x1200009d4      44869c27       addiu gp, gp, -31164
  ╎╎╎╎╎╎╎   0x1200009d8      2de09f03       daddu gp, gp, ra
  ╎╎╎╎╎╎╎   0x1200009dc      2df80000       move ra, zero
  ╎╎╎╎╎╎╎   0x1200009e0      208084df       ld a0, -32736(gp)
  ╎╎╎╎╎╎╎   0x1200009e4      0000a5df       ld a1, 0(sp)
  ╎ ╎└────< 0x1200009e8      0800a667       daddiu a2, sp, 8
  ╎ ╎╎╎╎╎   0x1200009ec      f0ff0124       li at, -16
```
### Steps to reproduce the behavior 
```
: e asm.arch =mips.gnu
: V
: p 
```
Disassembly panel

### PR DESCRIPTION
**Pull request by junchao-loongson**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

We should judge Bit Endianness before analyzing

**Test plan**

Will be tested after fixing the load instruction later because it seems difficult to test separately

**Closing issues**

closes #17413

### CODE REVIEW

**Comment by XVilka** [comment]
Please use `r_read_ble32()` instead, see [libr/include/r_endian.h](https://github.com/radareorg/radare2/blob/master/libr/include/r_endian.h).

**Comment by junchao-loongson** [comment]
> Please use `r_read_ble32()` instead, see [libr/include/r_endian.h](https://github.com/radareorg/radare2/blob/master/libr/include/r_endian.h).

We need to use the buf array for subsequent code work

**Comment by XVilka** [comment]
Just cast array to `ut8 *` and that's it.

File: libr/anal/p/anal_mips_gnu.c [hunk 1]
```diff
@@ -37,13 +38,18 @@ static int mips_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *b, int len, R
...
-	// eprintf ("MIPS: %02x %02x %02x %02x (after endian: big=%d)\n", b[0], b[1], b[2], b[3], anal->big_endian);
+	// eprintf ("MIPS: %02x %02x %02x %02x (after endian: big=%d)\n", buf[0], buf[1], buf[2], buf[3], anal->big_endian);
 	if (opcode == 0) {
 		op->type = R_ANAL_OP_TYPE_NOP;
 		return oplen;
 	}
-
-	optype = (b[0]>>2);
+	
+	buf[0] = opcode>>24;
```
Comment (XVilka) [inline libr/anal/p/anal_mips_gnu.c:47]: Just use `r_swap_ut32(ut32 val)` instead of manually doing it.

**Pull request merged**